### PR TITLE
feat: add support to set ICL as targets in logs routing

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/cloud_logs/main.tf [hunk 1]
```diff
@@ -106,3 +107,55 @@ resource "ibm_logs_outgoing_webhook" "en_integration" {
...
+  special = false
+}
+
+resource "ibm_logs_router_tenant" "logs_router_tenant_instance" {
+  name = var.logs_routing_tenant_name != null ? var.logs_routing_tenant_name : "${var.region}-${random_string.random_tenant_suffix.result}"
+  targets {
+    log_sink_crn = ibm_resource_instance.cloud_logs.crn
+    name         = local.instance_name
+    parameters {
+      host = "${ibm_resource_instance.cloud_logs.guid}.ingress.${var.region}.logs.cloud.ibm.com"
```
Comment (ocofaigh) [inline modules/cloud_logs/main.tf:157]: Is this not a public host? We probably want to use private by default, but I guess we should also optionally allow user to public incase its not a VRF enabled account.

File: modules/cloud_logs/variables.tf [hunk 2]
```diff
@@ -122,3 +122,29 @@ variable "existing_en_instances" {
...
 }
+
+##############################################################################
+# Logs Routing
+##############################################################################
+
+variable "enable_cloud_logs_as_target" {
+  description = "Whether to set IBM Cloud Logs be designated as a target for logs routing in a specific region."
+  type        = bool
+  default     = false
```
Comment (ocofaigh) [inline modules/cloud_logs/variables.tf:133]: Is there a reason why we default to false here instead of true?

File: variables.tf [hunk 3]
```diff
@@ -461,4 +461,23 @@ variable "cloud_logs_data_storage" {
   }
   description = "A logs data bucket and a metrics bucket in IBM Cloud Object Storage to store your IBM Cloud Logs data for long term storage, search, analysis and alerting."
 }
+
+variable "enable_cloud_logs_as_target" {
+  description = "Whether to set IBM Cloud Logs be designated as a target for logs routing in a specific region."
+  type        = bool
+  default     = false
```
Comment (ocofaigh) [inline variables.tf:468]: same question as above, why not default to true?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
A few comments

File: variables.tf (hunk 3 above)
Comment (maheshwarishikha) [reply variables.tf]: Not sure if there was any specific reason. Changed it to true.

File: modules/cloud_logs/variables.tf (hunk 2 above)
Comment (maheshwarishikha) [reply modules/cloud_logs/variables.tf]: Not sure if there was any specific reason. Changed it to true.

File: modules/cloud_logs/main.tf (hunk 1 above)
Comment (maheshwarishikha) [reply modules/cloud_logs/main.tf]: For this, I used service_endpoints as reference, if Cloud logs is setup with public endpoint then use public endpoint or else private. 
Let me know if any concern.

**Comment by maheshwarishikha** [comment]
This PR removes the hard coded region of cloud logs from example
`cloud_logs_region           = "eu-es"`
Because of which upgrade test is failing.

image

For now, it is safe to upgrade test, hence skipping.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
There are a few changes we need to make here. Im going to push a commit wih some updates. But bare in mind, the current code is only creating 1 tenant in whatever region the provider is set to. And in order to get platform logs from services in all regions, you need to create a tenant in every region.
It seems this is not yet supported in the provider, but I see its coming in https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5634
I am pushing to see if we can get this merged and a hotfix generated, but in parallel I'll proceed with updating the code in this PR directly which would only support 1 tenant in 1 region.

**Comment by ocofaigh** [comment]
The provider team might be creating a patch release for us when https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5634 is merged. If this is the case, I would like to wait for that instead of merging the code with the limitation

**Review by ocofaigh — CHANGES_REQUESTED** [review]
As we discovered, only public ingress endpoints are supported when configuring a target to an IBM Cloud Logs instance, so we need to remove the `use_private_endpoint_logs_routing` variable and always use public for now

File: modules/cloud_logs/main.tf [hunk 4]
```diff
@@ -165,15 +163,14 @@ resource "ibm_logs_router_tenant" "logs_router_tenant_instances" {
   # until provider supports passing region to this resource (coming in https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5634),
   # the for_each will only ever include the provider region
 
-  # for_each = contains(var.logs_routing_tenant_regions, "*") ? toset(data.ibm_is_regions.regions.regions[*].name) : var.logs_routing_tenant_regions
```
Comment (ocofaigh) [inline modules/cloud_logs/main.tf]: oh @Aashiq-J I left that that code comment here on purpose, so we know to swithc to it when the provider support comes. You might want to add it back

**Comment by ocofaigh** [comment]
New provider version is coming today with the fix we need. I can update this PR once its out

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**